### PR TITLE
Add support for secure boot and FDE

### PR DESCRIPTION
**Pull request by alexgg**
Changelog-entry: Add support for secure boot and FDE to Generic x86_64
Signed-off-by: Michal Toman <michalt@balena.io>

### CODE REVIEW

File: layers/meta-balena-genericx86/recipes-core/images/balena-image-flasher.bbappend [hunk 1]
```diff
@@ -2,7 +2,10 @@ include balena-image.inc
 
 BALENA_BOOT_PARTITION_FILES_append = " \
     grub.cfg_external:/EFI/BOOT/grub.cfg \
+    grub.cfg_external.sig:/EFI/BOOT/grub.cfg.sig \
     grub.cfg_external:/grub/grub.cfg \
     grub.cfg_internal: \
+    grub.cfg_internal.sig: \
+    grub.cfg_internal_luks: \
+    grub.cfg_internal_luks.sig: \
```
[the remark this answers is not in the document]
Comment (alexgg) [reply layers/meta-balena-genericx86/recipes-core/images/balena-image-flasher.bbappend:10]: Done

File: layers/meta-balena-genericx86/recipes-core/images/balena-image.bbappend [hunk 2]
```diff
@@ -2,6 +2,7 @@ include balena-image.inc
 
 BALENA_BOOT_PARTITION_FILES_append = " \
 	grub.cfg_internal:/EFI/BOOT/grub.cfg \
+	grub.cfg_internal.sig:/EFI/BOOT/grub.cfg.sig \
```
[the remark this answers is not in the document]
Comment (alexgg) [reply layers/meta-balena-genericx86/recipes-core/images/balena-image.bbappend]: Done

**Comment by alexgg** [comment]
@balena-ci rebase

**Comment by mtoman** [comment]
Let's also ship the `balena-keys` directory in the boot partition. The contents are created by meta-balena and we can deal with the formats there.

I am thinking about `balena-keys:/balena-keys/` in `layers/meta-balena-genericx86/recipes-core/images/balena-image.inc`

**Comment by alexgg** [comment]
> Let's also ship the `balena-keys` directory in the boot partition. The contents are created by meta-balena and we can deal with the formats there.
> 
> I am thinking about `balena-keys:/balena-keys/` in `layers/meta-balena-genericx86/recipes-core/images/balena-image.inc`

@mtoman every device that configures secure boot will want the keys in the boot partition, right? So I think it's a meta-balena change too.

**Comment by alexgg** [comment]
@balena-ci rebase

**Comment by alexgg** [comment]
@resin-jenkins retest this please

**Comment by klutchell** [comment]
@balena-ci rebase

**Comment by alexgg** [comment]
@resin-jenkins retest this please

**Comment by alexgg** [comment]
@balena-ci I self-certify!

**Pull request merged**